.circleci/config.yml

```diff
@@ -0,0 +1,452 @@
+          # Use the latest mongo that Stouts supports
+          command: pyenv exec ansible-playbook -i "localhost," -c local examples/girder-dev-environment/site.yml --extra-vars="{'girder_version':'$CIRCLE_BRANCH', 'girder_repo':'$CIRCLE_REPOSITORY_URL', 'girder_path':'$CIRCLE_WORKING_DIRECTORY/girder', 'girder_clone':false, 'girder_update':false, 'girder_web':false, 'mongodb_version':'3.2.4'}"
+          working_directory: girder/devops/ansible
+          environment:
+            ANSIBLE_ROLES_PATH: roles:examples/girder-dev-environment/roles
+      - run:
+          name: Activate virtualenv
+          command: echo '. /home/circleci/.virtualenvs/girder/bin/activate' >> $BASH_ENV
+      - run:
+          name: Install all pip modules
+          # sftp and python client aren't installed by the ansible example
+          command: pip install -e .[plugins,sftp] clients/python -r requirements-dev.txt
+          working_directory: girder
+      - run:
+          name: Create Girder build directory
+          command: mkdir girder_build
+      - run:
+          name: Create various subdirectories
+          command: mkdir -p build/test/coverage/web_temp && mkdir -p build/test/coverage/web
+          working_directory: girder
+      - run:
+          name: Build Girder web client with coverage turned on
+          command: girder-install web --all-plugins --dev | cat
+      - run:
+          name: CMake
+          command: cmake ../girder -DPYTHON_VERSION=3.4 -DPYTHON_EXECUTABLE=/home/circleci/.virtualenvs/girder/bin/python -DVIRTUALENV_EXECUTABLE=/home/circleci/.virtualenvs/girder/bin/virtualenv
+          working_directory: girder_build
+          environment:
+            JASMINE_TIMEOUT: 15000
+      - run:
+          name: make
+          command: make -j 3
+          working_directory: girder_build
+      - run:
+          name: CTest
+          command: ctest -j 3 -VV -L girder_integration
+          working_directory: girder_build
+          environment:
+            JASMINE_TIMEOUT: 15000
+      - run:
+          name: Rename javascript coverage files to avoid conflicts
+          command: for file in *.cvg; do mv "$file" "${file%.cvg}integration.cvg"; done
+          working_directory: girder/build/test/coverage/web_temp
+      - persist_to_workspace:
+          root: /home/circleci/project
+          paths:
+            - girder/.coverage.*
+            - girder/build/test/coverage/web_temp/*.cvg
+
+  py3_coverage:
+    docker:
+      # This image uses the oldest version of many dependencies
+      - image: girder/girder_test_py3:latest
+      # Use the latest MongoDB
+      - image: circleci/mongo:3.4
+    working_directory: /home/circleci/project # as $CIRCLE_WORKING_DIRECTORY
+    steps:
+      - attach_workspace:
+          at: /home/circleci/project
+      - run:
+          name: Activate virtual environment
+          command: echo "source $CIRCLE_WORKING_DIRECTORY/girder_env/bin/activate" >> $BASH_ENV
```

Comment:
Can we use `.` or `source` uniformly? I heard @brianhelba has some opinions on this.

Reply:
Sure.  I'd recommend never using `source` and always using `.`, as it is a cross-shell method.